builtin/providers/aws/resource_aws_rds_cluster.go

```diff
@@ -499,6 +511,16 @@ func resourceAwsRDSClusterRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("reader_endpoint", dbc.ReaderEndpoint)
 	d.Set("replication_source_identifier", dbc.ReplicationSourceIdentifier)
 
+	var roles []string
+	for _, r := range dbc.AssociatedRoles {
+		if *r.Status == "ACTIVE" {
+			roles = append(roles, *r.RoleArn)
```

Comment:
So the issue I faced here is that the UI only shows ACTIVE roles but the API returns the full history of the role associations - so i decided that we would only grab the attached ones

Maybe this isn't correct?

P.